src/value.cpp

```diff
@@ -1154,13 +1154,13 @@ namespace Exiv2 {
         char plusMinus = '+';
         if (time_.tzHour < 0 || time_.tzMinute < 0) plusMinus = '-';
 
-        const int wrote = sprintf(temp,
+        const int wrote = snprintf(temp, sizeof(temp),
```

Comment:
You should pass it `sizeof(temp) - 1` as `snprinft` will not terminate your string with a `\0` if the buffer is too small. We should also `memset` temp, as otherwise the last element of `temp` will be random.

Reply:
I'm pretty sure in this case we have to pass `12` here (or `sizeof(temp)`) and then the function is returning `11`. I spent some time debugging this code and running the tests I added in #347 .

This is what the `snprintf` documentation says:

```
Writes the results to a character string buffer. At most bufsz - 1 characters are written. The resulting character string will be terminated with a null character, unless bufsz is zero. If bufsz is zero, nothing is written and buffer may be a null pointer, however the return value (number of bytes that would be written) is still calculated and returned.
```
So, `bufzs` needs to be 12 but at the end only 11 bytes are written and the last one is assigned to `\0`.

Reply:
Ah, you are right. I didn't read the manpage carefully enough (where did you get the pasted documentation from? I haven't found it in the man page). It should work the way it is.